gratipay/models/participant/__init__.py

```diff
@@ -1135,8 +1135,11 @@ def change_username(self, suggested):
 
         lowercased = suggested.lower()
 
-        if lowercased in gratipay.RESTRICTED_USERNAMES:
-            raise UsernameIsRestricted(suggested)
+        # Don't allow any username which is the name of a
+        # file existing on the web_root folder.
+        for ext in ['', '.spt', '.ico', '.txt', '.osdd']:
+            if (lowercased + ext) in gratipay.RESTRICTED_USERNAMES:
+                raise UsernameIsRestricted(suggested)
```

Comment:
Hrm, not easily. Dispatch is embedded in Gratipay's [algorithm](https://github.com/gratipay/gratipay.com/blob/cecb6ce265c7dba325cf4561a5acef948feb727e/gratipay/main.py#L97-L138), and it seems that we'd want to test the real algorithm.